src/runtime/tests/refiner-test.ts

```diff
@@ -17,338 +17,362 @@ import {Schema} from '../schema.js';
+    const data = {
+      num: 6
+    };
+    const res = ref.validateData(data);
+    assert.strictEqual(res, data.num === 0 || data.num === 6);
+  }));
+  it('Refines data given a refinement', () => {
+    const manifestAst = parse(`
+      particle Foo
+        input: reads Something {num: Number } [(num/-3 < 0 and num*num == 36) or num == 0]
+    `);
+    const typeData = {'num': 'Number'};
+    const ref = Refinement.fromAst(manifestAst[0].args[0].type.refinement, typeData);
+    const data = {
+      num: 6
+    };
+    const res = ref.validateData(data);
+    assert.strictEqual(res, data.num === 0 || data.num === 6);
+  });
+  it('Throws error when field name not found in field refinement', Flags.withFieldRefinementsAllowed(async () => {
+    const manifestAst = parse(`
+      particle Foo
+        input: reads Something {num: Number [ num < num2 ] }
+    `);
+    const typeData = {'num': 'Number'};
+    assert.throws(() => {
+      Refinement.fromAst(manifestAst[0].args[0].type.fields[0].type.refinement, typeData);
+    }, `Unresolved field name 'num2' in the refinement expression.`);
+  }));
+  it('Throws error when field name not found in refinement', () => {
+    const manifestAst = parse(`
+      particle Foo
+        input: reads Something {num: Number } [num < num2]
+    `);
+    const typeData = {'num': 'Number'};
+    assert.throws(() => {
+      Refinement.fromAst(manifestAst[0].args[0].type.refinement, typeData);
+    }, `Unresolved field name 'num2' in the refinement expression.`);
   });
-    it('Throws error when field name not found 1', Flags.withFieldRefinementsAllowed(async () => {
-        assert.throws(() => {
-            const manifestAst = parse(`
-                particle Foo
-                    input: reads Something {num: Number [ num < num2 ] }
-            `);
-            const typeData = {'num': 'Number'};
-            const ref = Refinement.fromAst(manifestAst[0].args[0].type.fields[0].type.refinement, typeData);
-            const data = {
-                num: 6,
-            };
-            ref.validateData(data);
-        }, `Unresolved field name 'num2' in the refinement expression.`);
-    }));
-    it('Throws error when field name not found 2', () => {
-      assert.throws(() => {
-          const manifestAst = parse(`
-              particle Foo
-                  input: reads Something {num: Number } [num < num2]
-          `);
-          const typeData = {'num': 'Number'};
-          const ref = Refinement.fromAst(manifestAst[0].args[0].type.refinement, typeData);
-          const data = {
-              num: 6,
-          };
-          ref.validateData(data);
-      }, `Unresolved field name 'num2' in the refinement expression.`);
+  it('Throws error when field name not found in query refinement', () => {
+    const manifestAst = parse(`
+      particle Foo
+        input: reads Something {num: Number } [num < ?]
+    `);
+    const typeData = {'num': 'Number'};
```

Comment:
This seems like it could be shared between most of the tests.

Reply:
Most but not all, could be confusing? Will leave it for now.